### PR TITLE
Set the GCP compatibility label based on environment in config helper.

### PR DESCRIPTION
**Pull request by bmoyles0117**
**What this PR does / why we need it**:
Adds support for adjusting the record label sent to the Google Cloud Logging API to indicate which resource type the logs should be written against.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2155,6 +2155,12 @@ EOF
     update-event-exporter ${event_exporter_yaml}
     fluentd_gcp_version="${FLUENTD_GCP_VERSION:-0.2-1.5.28-1}"
     sed -i -e "s@{{ fluentd_gcp_version }}@${fluentd_gcp_version}@g" "${fluentd_gcp_yaml}"
+    if [[ "${LOGGING_STACKDRIVER_RESOURCES:-}" == "new" ]]; then
+      fluentd_k8s_compatibility="true"
+    else
+      fluentd_k8s_compatibility="false"
+    fi
+    sed -i -e "s@{{ fluentd_k8s_compatibility }}@${fluentd_k8s_compatibility}@g" "${fluentd_gcp_yaml}"
```
Comment (crassirostris) [inline cluster/gce/gci/configure-helper.sh]: I really don't like doing all this inline, please extract it to separate function, maybe with the previous couple of lines, to make a logical block like `setup-fluentd`

**Review by crassirostris — COMMENTED** [review]
LGTM, one nit

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -2155,6 +2155,12 @@ EOF
     update-event-exporter ${event_exporter_yaml}
     fluentd_gcp_version="${FLUENTD_GCP_VERSION:-0.2-1.5.28-1}"
     sed -i -e "s@{{ fluentd_gcp_version }}@${fluentd_gcp_version}@g" "${fluentd_gcp_yaml}"
+    if [[ "${LOGGING_STACKDRIVER_RESOURCES:-}" == "new" ]]; then
```
Comment (x13n) [inline cluster/gce/gci/configure-helper.sh]: I think we need to add this to PROVIDER_VARS so that it is set in the cluster.

Comment (bmoyles0117) [reply cluster/gce/gci/configure-helper.sh]: I don't see `HEAPSTER_USE_OLD_STACKDRIVER_RESOURCES` or `HEAPSTER_USE_NEW_STACKDRIVER_RESOURCES` in the `PROVIDER_VARS`, are you sure we need it, if we do, are we missing it for these heapster variables?

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (bmoyles0117) [reply cluster/gce/gci/configure-helper.sh]: PTAL

File: cluster/gce/gci/configure-helper.sh [hunk 3]
```diff
@@ -2148,17 +2168,7 @@ EOF
   fi
   if [[ "${ENABLE_NODE_LOGGING:-}" == "true" ]] && \
      [[ "${LOGGING_DESTINATION:-}" == "gcp" ]]; then
-    setup-addon-manifests "addons" "fluentd-gcp"
-    local -r event_exporter_yaml="${dst_dir}/fluentd-gcp/event-exporter.yaml"
-    local -r fluentd_gcp_yaml="${dst_dir}/fluentd-gcp/fluentd-gcp-ds.yaml"
-    local -r fluentd_gcp_configmap_yaml="${dst_dir}/fluentd-gcp/fluentd-gcp-configmap.yaml"
-    update-event-exporter ${event_exporter_yaml}
```
Comment (crassirostris) [inline cluster/gce/gci/configure-helper.sh]: `event-exporter` is not fluentd

File: cluster/gce/gci/configure-helper.sh [hunk 4]
```diff
@@ -2148,17 +2168,7 @@ EOF
   fi
   if [[ "${ENABLE_NODE_LOGGING:-}" == "true" ]] && \
      [[ "${LOGGING_DESTINATION:-}" == "gcp" ]]; then
-    setup-addon-manifests "addons" "fluentd-gcp"
```
Comment (crassirostris) [inline cluster/gce/gci/configure-helper.sh]: I think this is also not about setting up fluentd parameters, but setting up clusters parameters to use fluentd

File: cluster/gce/gci/configure-helper.sh [hunk 5]
```diff
@@ -2013,6 +2013,26 @@ function setup-coredns-manifest {
   sed -i -e "s@{{ *pillar\['service_cluster_ip_range'\] *}}@${SERVICE_CLUSTER_IP_RANGE}@g" "${coredns_file}"
 }
 
+function setup-fluentd {
+  setup-addon-manifests "addons" "fluentd-gcp"
+  local -r event_exporter_yaml="${dst_dir}/fluentd-gcp/event-exporter.yaml"
```
Comment (crassirostris) [inline cluster/gce/gci/configure-helper.sh]: `dst_dir` is a local variable of the function you extracted this from

Please pass variables via function parameters

**Review by crassirostris — COMMENTED** [review]
> I don't see `HEAPSTER_USE_OLD_STACKDRIVER_RESOURCES` or `HEAPSTER_USE_NEW_STACKDRIVER_RESOURCES` in the `PROVIDER_VARS`, are you sure we need it, if we do, are we missing it for these heapster variables?

The latter

File: cluster/gce/gci/configure-helper.sh (hunk 5 above)
Comment (bmoyles0117) [reply cluster/gce/gci/configure-helper.sh]: If this causes this to be broken, the same could apply to `setup-coredns-manifest`, and potentially other helpers that exist throughout this file.

File: cluster/gce/gci/configure-helper.sh (hunk 4 above)
Comment (bmoyles0117) [reply cluster/gce/gci/configure-helper.sh]: Moved.

File: cluster/gce/gci/configure-helper.sh (hunk 3 above)
Comment (bmoyles0117) [reply cluster/gce/gci/configure-helper.sh]: Moved.

**Review by bmoyles0117 — COMMENTED** [review]
PTAL

**Comment by crassirostris** [comment]
LGTM, please rebase

Deferring applying label to @x13n

**Comment by bmoyles0117** [comment]
Rebased with all comments addressed.

**Comment by x13n** [comment]
/assign @MaciekPytel 
Can you take a look? We'd like to have this switch in 1.10 branch, too, so we'll need a second PR with cherrypick later.

**Comment by x13n** [comment]
Just synced with @bmoyles0117 - unfortunately this needs more work than just setting one label. There will be a separate PR for this.
/lgtm cancel
/close

**Pull request closed without merging**